### PR TITLE
Atanas kisyov

### PR DESCRIPTION
**Pull request by AtanasKisyov**

### CODE REVIEW

File: Carwash/orders/tests.py [hunk 1]
```diff
@@ -1,3 +1,32 @@
...
+        data = {
+            'car_wash': self.order.car_wash,
+            'user': self.order.user,
+            'order_current_status': self.order.order_current_status,
+            'execution': self.order.execution,
+        }
+        user_response = self.client.post('127.0.0.1:8000', data=user)
+        data_response = self.client.post(reverse('orders'), data=data)
+        order = Order.objects.all()
+        print(user_response)
```
Comment (strangerok1507) [inline Carwash/orders/tests.py]: test isn't valid. it doesn't check anything

File: Carwash/orders/factories.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+from users.models import Accounts, Users
+
+
+class AccountFactory(factory.django.DjangoModelFactory):
+
+    class Meta:
+        model = Accounts
+        django_get_or_create = ('email',)
+
+    email = 'arthur.dent@the-guide.com'
```
Comment (strangerok1507) [inline Carwash/orders/factories.py]: use factory for it
email = factory.Faker('email')

File: Carwash/orders/factories.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    user_location = 'Earth'
+    account = factory.SubFactory(AccountFactory)
+
+
+class CarWashFactory(factory.django.DjangoModelFactory):
+
+    class Meta:
+        model = CarWashes
+    
+    car_wash_name = 'Car Wash at the end of the Universe'
```
Comment (strangerok1507) [inline Carwash/orders/factories.py]: use faker for these fields

**Review by strangerok1507 — CHANGES_REQUESTED** [review]

File: Carwash/orders/tests.py (hunk 1 above)
Comment (AtanasKisyov) [reply Carwash/orders/tests.py]: I committed the tests by mistake... I will commit the proper ones later today.

File: Carwash/orders/tests.py [hunk 4]
```diff
@@ -1,3 +1,47 @@
...
+        self.car_wash = factories.CarWashFactory.create()
+        self.order = factories.OrderFactory.create()
+        
+    def test_successful_create_data(self):
+        """Test if object is created and saved"""
+
+        data = Order.objects.all()
+        self.assertIsNotNone(data)
+    
+    def test_successful_delete_data(self):
```
Comment (strangerok1507) [inline Carwash/orders/tests.py]: you don't send request to api endpoint

File: Carwash/orders/tests.py [hunk 5]
```diff
@@ -1,3 +1,47 @@
...
+        """Test if object is deleted"""
+
+        instance = Order.objects.first()
+        instance.delete()
+        instance = Order.objects.all()
+        self.assertEquals(len(instance), 0)
+
+    def test_successful_update_data(self):
```
Comment (strangerok1507) [inline Carwash/orders/tests.py]: you don't send request to api endpoint

**Review by strangerok1507 — CHANGES_REQUESTED** [review]

File: Carwash/orders/tests.py (hunk 4 above)
Comment (AtanasKisyov) [reply Carwash/orders/tests.py]: Tests are added

File: Carwash/orders/tests.py (hunk 5 above)
Comment (AtanasKisyov) [reply Carwash/orders/tests.py]: Tests are added

File: Carwash/orders/tests.py (hunk 4 above)
Comment (strangerok1507) [reply Carwash/orders/tests.py]: it's cool. but remove tests where you don't send request

File: Carwash/orders/tests.py (hunk 5 above)
Comment (strangerok1507) [reply Carwash/orders/tests.py]: it's cool. but remove tests where you don't send request

File: Carwash/orders/tests.py [hunk 6]
```diff
@@ -1,3 +1,139 @@
...
+
+        posted_object_id = post.json()['id']
+        self.order_object['order_current_status'] = 'Changed'
+        
+        self.client.put(
+            f'{self.order_url}{posted_object_id}/',
+            self.order_object,
+        )
+
+        changed_object = self.client.get(
```
Comment (strangerok1507) [inline Carwash/orders/tests.py]:
```suggestion
        response = self.client.get(
```

Comment (strangerok1507) [reply Carwash/orders/tests.py]: and fix other lines in this test

File: Carwash/orders/tests.py [hunk 7]
```diff
@@ -1,3 +1,139 @@
...
+        """Test if object is deleted successfully with delete request"""
+
+        post = self.client.post(
+            self.order_url,
+            self.order_object,
+        )
+
+        posted_object_id = post.json()['id']
+
+        delete = self.client.delete(
```
Comment (strangerok1507) [inline Carwash/orders/tests.py]:
```suggestion
        response = self.client.delete(
```

File: Carwash/orders/tests.py [hunk 8]
```diff
@@ -1,3 +1,139 @@
...
+
+        post = self.client.post(
+            self.order_url,
+            self.order_object,
+        )
+
+        posted_object_id = post.json()['id']
+        self.order_object['order_current_status'] = 'Changed'
+        
+        self.client.put(
```
Comment (strangerok1507) [inline Carwash/orders/tests.py]:
```suggestion
        response = self.client.put(
```
and check status of response

**Review by strangerok1507 — CHANGES_REQUESTED** [review]

**Review by strangerok1507 — APPROVED** [review]

**Comment by strangerok1507** [comment]
Looks good to me)

**Pull request merged**